test/gtest/ucp/test_ucp_atomic.cc

```diff
@@ -102,6 +119,28 @@ void test_ucp_atomic::nb_post(entity *e,  size_t max_size, void *memheap_addr,
     *(T*)&expected_data[0] = atomic_op_val<T, OP>(val, prev);
 }
 
+template <typename T, ucp_atomic_op_t OP>
+void test_ucp_atomic::nbx_post(entity *e,  size_t max_size, void *memheap_addr,
+                               ucp_rkey_h rkey, std::string& expected_data)
+{
+    ucs_status_t status;
+    T val, prev;
+
+    prev   = *(T*)memheap_addr;
+    val    = (T)ucs::rand() * (T)ucs::rand();
+
+    status = test_ucp_atomic::ucp_atomic_post_nbx<T>(e->ep(), OP, val,
+                                                     memheap_addr, rkey);
+
+    if (status == UCS_INPROGRESS) {
```

Comment:
if status is INPROGRESS, where do we wait/release the request?

Reply:
we wait in flush, request is released in `ucp_atomic_post_nbx`, if no `reply_buffer` provided

Reply:
asking differently: in which case status can be UCS_INPROGRESS here?

Reply:
it can't